### ISSUE TITLE
seq_linter should catch seq(length()), redirecting to seq_along()

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1467]
```r
seq(0)
# [1] 1 0
```

i.e., `seq(length(x))` is subject to the same unexpected/error-prone edge-case behavior as other usages caught by `seq_linter()`

**Comment by IndrajeetPatil** [comment]
I think this already works? We just need to add tests for this scenario.

``` r
library(lintr)

lint(
  text = "1:seq(length(x))",
  linters = seq_linter()
)
#> <text>:1:1: warning: [seq_linter] 1:seq(...) is likely to be wrong in the empty edge case. Use seq_len() instead.
#> 1:seq(length(x))
#> ^~~~~~~~~~~~~~~~
```

<sup>Created on 2022-07-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by IndrajeetPatil** [comment]
It does produce a false positive though:

``` r
library(lintr)

lint(
  text = "1:seq_len(length(x))",
  linters = seq_linter()
)
#> <text>:1:1: warning: [seq_linter] 1:seq_len(...) is likely to be wrong in the empty edge case. Use seq_len() instead.
#> 1:seq_len(length(x))
#> ^~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2022-07-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by MichaelChirico** [comment]
NB: there's no '1:' in my example

**Comment by IndrajeetPatil** [comment]
Oops! Gotcha. This indeed doesn't produce a lint:

``` r
library(lintr)

lint(
  text = "seq(length(mtcars))",
  linters = seq_linter()
)
```

<sup>Created on 2022-07-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by IndrajeetPatil** [comment]
Is this better?

With #1468:

``` r
library(lintr)

lint(
  text = "seq(length(mtcars))",
  linters = seq_linter()
)
#> <text>:1:1: warning: [seq_linter] seq(length(...)) is likely to be wrong in the empty edge case. Use seq(seq_along(...)) instead.
#> seq(length(mtcars))
#> ^~~~~~~~~~~~~~~~~~~

lint(
  text = "seq(nrow(mtcars))",
  linters = seq_linter()
)
#> <text>:1:1: warning: [seq_linter] seq(nrow(...)) is likely to be wrong in the empty edge case. Use seq(seq_len(...)) instead.
#> seq(nrow(mtcars))
#> ^~~~~~~~~~~~~~~~~
```

<sup>Created on 2022-07-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
